packages/react/select/src/Select.tsx

```diff
@@ -1031,7 +1042,7 @@ const SelectScrollUpButton = React.forwardRef<
   const [canScrollUp, setCanScrollUp] = React.useState(false);
   const composedRefs = useComposedRefs(forwardedRef, contentContext.onScrollButtonChange);
 
-  React.useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
I believe these were the only thing needed to make it go green and the reason I saw the jumps and misalignments in React 18.